localstack/services/events/target.py

```diff
@@ -0,0 +1,317 @@
+import json
+import logging
+import uuid
+from abc import ABC, abstractmethod
+
+from localstack.aws.api.events import (
+    Arn,
+    Target,
+)
+from localstack.aws.connect import ServiceLevelClientFactory, connect_to
+from localstack.utils import collections
+from localstack.utils.aws.arns import (
+    firehose_name,
+    parse_arn,
+    sqs_queue_url_for_arn,
+)
+from localstack.utils.aws.client_types import ServicePrincipal
+from localstack.utils.strings import to_bytes
+from localstack.utils.time import now_utc
+
+LOG = logging.getLogger(__name__)
+
+
+def connect_with_role(role_arn: str, region_name: str) -> ServiceLevelClientFactory:
+    """Connect to an internal botoclient factory using an assumed role."""
+    client = connect_to.with_assumed_role(
+        role_arn=role_arn, service_principal=ServicePrincipal.events, region_name=region_name
+    )
+    return client
+
+
+class TargetService(ABC):
```

Comment:
Not really happy with the naming here. Could we perhaps get rid of the `Service`? I am aware of the overlap with `Target` from the API spec then, but we could maybe call it `RuleTarget` or `EventTarget` or even `TargetPublisher` `TargetSender` or something the likes, but I am not a fan of `TargetService`, personally.

Since this is more a naming debate, please do not let this hold back the merge.

Reply:
+1 for `TargetSender` in line with EventBridge Pipes
It implements a send method, so sender seems appropriate

Reply:
for targets it indeed makes much more sense to call it sender, renamed in: https://github.com/localstack/localstack/pull/10613/commits/344f54cfdc3e966e5e97aa81cece34c13b7b68e0